src/test/java/org/pesmypetcare/webservice/service/MedicationServiceTest.java

```diff
@@ -0,0 +1,152 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class MedicationServiceTest {
+    private static List<Map<List<String>, Object>> MedicationList;
+    private static MedicationEntity MedicationEntity;
+    private static String owner;
+    private static String petName;
+    private static String dateName;
+    private static String date;
+    private static String date2;
+    private static String field;
+    private static int value;
+
+    @Mock
+    private MedicationDao MedicationDao;
+
+    @InjectMocks
+    private MedicationService service = new MedicationServiceImp();
+
+    @BeforeAll
+    public static void setUp() {
+        MedicationList = new ArrayList<>();
+        MedicationEntity = new MedicationEntity();
+        dateName = "2020-02-13/Cloroform";
+        date = "2020-02-13";
+        date2 = "2021-02-13";
+        owner = "Pepe05";
+        petName = "Camper";
+        field = "quantity";
+        value = 3;
+    }
+
+    @Test
+    public void shouldReturnNothingWhenMedicationCreated() {
+        service.createMedication(owner, petName, dateName, MedicationEntity);
+        verify(MedicationDao).createMedication(isA(String.class), isA(String.class), isA(String.class), isA(MedicationEntity.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenMedicationDeleted() {
+        service.deleteByDateAndName(owner, petName, dateName);
+        verify(MedicationDao).deleteByDateAndName(isA(String.class), isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenAllMedicationsDeleted() throws DatabaseAccessException {
+        service.deleteAllMedications(owner, petName);
+        verify(MedicationDao).deleteAllMedications(isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenAllMedicationDeleteFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(MedicationDao).deleteAllMedications(any(String.class), isA(String.class));
+            service.deleteAllMedications(owner, petName);
+        }, "Should return a database access exception when a Medication deletion fails");
+    }
+
+    @Test
+    public void shouldReturnMedicationEntityWhenMedicationRetrieved() throws DatabaseAccessException {
+        when(MedicationDao.getMedicationData(owner, petName, dateName)).thenReturn(MedicationEntity);
+        MedicationEntity Medication = service.getMedicationData(owner, petName, dateName);
+        assertSame(MedicationEntity, Medication, "Should return a Medication entity");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetMedicationRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(MedicationDao).getMedicationData(any(String.class), any(String.class),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 126).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)